File: backend/app/schemas/platforms.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class PlatformBase(BaseModel):
    name: str = Field(..., max_length=100, description="平台名称")
    code: str = Field(..., max_length=50, description="平台代码")
    icon: Optional[str] = Field(None, max_length=255, description="平台图标")
    description: Optional[str] = Field(None, description="平台描述")
    is_active: int = Field(default=1, description="是否启用")
    sort_order: int = Field(default=0, description="排序")
    config: Optional[dict] = Field(None, description="平台配置")


class PlatformCreate(PlatformBase):
    pass


class PlatformUpdate(BaseModel):
    name: Optional[str] = None
    code: Optional[str] = None
    icon: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[int] = None
    sort_order: Optional[int] = None
    config: Optional[dict] = None


class PlatformResponse(PlatformBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/services/platforms.ts
import request from '@/utils/request'

export interface Platform {
  id: number
  name: string
  code: string
  icon?: string
  description?: string
  is_active: number
  sort_order: number
  config?: Record<string, any>
  created_at: string
  updated_at: string
}

export interface PlatformShop {
  id: number
  name: string
  platform_id: number
  account: string
  status: string
  manager_id?: number
  created_at: string
  updated_at: string
}

// 获取平台列表
export const getPlatforms = (params?: { is_active?: number; skip?: number; limit?: number }) => {
  return request.get<Platform[]>('/platforms', { params })
}

// 获取指定平台
export const getPlatform = (id: number) => {
  return request.get<Platform>(`/platforms/${id}`)
}

// 获取平台下的店铺
export const getPlatformShops = (id: number) => {
  return request.get<PlatformShop[]>(`/platforms/${id}/shops`)
}

// 创建平台
export const createPlatform = (data: Partial<Platform>) => {
  return request.post<Platform>('/platforms', data)
}

// 更新平台
export const updatePlatform = (id: number, data: Partial<Platform>) => {
  return request.put<Platform>(`/platforms/${id}`, data)
}

// 删除平台
export const deletePlatform = (id: number) => {
  return request.delete(`/platforms/${id}`)
}
